zilliqa/src/transaction.rs

```diff
@@ -708,6 +763,25 @@ impl Log {
             _ => None,
         }
     }
+
+    pub fn hash(&self) -> Hash {
+        match self {
+            Log::Scilla(log) => Hash::compute([
+                log.event_name.as_bytes(),
```

Comment:
I don't think this hash is as much of a hash as you think - Hash::compute() does not (I think?) insert boundaries between its preimages, so a log "foo" with param "bar" hashes to the same as a log with "foo<hash_of_bar>"